packages/swing-store/src/snapStore.js

```diff
@@ -9,6 +9,45 @@ const pipe = promisify(pipeline);
 
 const { freeze } = Object;
 
+/**
+ *
+ * @param {import("fs").ReadStream | import("fs").WriteStream} stream
+ * @returns {Promise<void>}
+ */
+export const fsStreamReady = stream =>
+  new Promise((resolve, reject) => {
+    if (stream.destroyed) {
+      reject(new Error('Stream already destroyed'));
+      return;
+    }
+
+    if (!stream.pending) {
+      resolve();
+      return;
+    }
+
+    stream.destroyed;
```

Comment:
Does this have a desirable side effect not achieved on line 19?

Reply:
Yeah I saw that, I think it was a left over of bad copy paste, it's also in the original 😳